pinot-common/src/main/java/org/apache/pinot/common/lineage/SegmentLineage.java

```diff
@@ -108,7 +120,9 @@ public static SegmentLineage fromZNRecord(ZNRecord record) {
       String lineageId = listField.getKey();
       List<String> value = listField.getValue();
       Preconditions.checkState(value.size() == 4);
-      List<String> segmentsFrom = Arrays.asList(value.get(0).split(COMMA_SEPARATOR));
+      String segmentsFromStr = value.get(0);
+      List<String> segmentsFrom = (segmentsFromStr == null || segmentsFromStr.length() == 0) ? new ArrayList<>()
+          : Arrays.asList(value.get(0).split(COMMA_SEPARATOR));
       List<String> segmentsTo = Arrays.asList(value.get(1).split(COMMA_SEPARATOR));
```

Comment:
Can `segmentsFromStr` be null?
Also avoid using the regex split
```suggestion
      List<String> segmentsFrom = Arrays.asList(StringUtils.split(value.get(0), ','));
      List<String> segmentsTo = Arrays.asList(StringUtils.split(value.get(1), ','));
```

Reply:
changed to `StringUtils.split()`.